api/src/pcapi/alembic/versions/20240613T085015_a3f3837659cb_add_future_offer_table.py

```diff
@@ -0,0 +1,66 @@
+"""Add future_offer table
+"""
+
+from alembic import op
+import sqlalchemy as sa
+
+from pcapi import settings
+
+
+# pre/post deployment: pre
+# revision identifiers, used by Alembic.
+revision = "a3f3837659cb"
+down_revision = "a909ecd976dc"
+branch_labels: tuple[str] | None = None
+depends_on: list[str] | None = None
+
+
+def upgrade() -> None:
+    op.create_table(
+        "future_offer",
+        sa.Column("id", sa.BigInteger(), autoincrement=True, nullable=False),
+        sa.Column("offerId", sa.BigInteger(), nullable=False),
+        sa.ForeignKeyConstraint(["offerId"], ["offer.id"], ondelete="CASCADE"),
+        sa.PrimaryKeyConstraint("id"),
+        sa.UniqueConstraint("offerId", name="unique_offer"),
+    )
+    with op.get_context().autocommit_block():
+        op.create_index(
+            op.f("ix_future_offer_offerId"),
+            "future_offer",
+            ["offerId"],
+            unique=False,
+            postgresql_concurrently=True,
+            if_not_exists=True,
+        )
+    op.add_column("offer", sa.Column("datePublished", sa.DateTime(), nullable=True))
+    with op.get_context().autocommit_block():
+        op.execute("SET SESSION statement_timeout='300s'")
+        op.create_index(
+            op.f("ix_offer_datePublished"),
+            "offer",
+            ["datePublished"],
+            unique=False,
+            postgresql_concurrently=True,
+            if_not_exists=True,
+        )
+        op.execute(f"SET SESSION statement_timeout={settings.DATABASE_STATEMENT_TIMEOUT}")
+
+
+def downgrade() -> None:
+    with op.get_context().autocommit_block():
+        op.drop_index(
```

Comment:
Le drop de la colonne va drop l'index automatiquement : 

> DROP COLUMN
This form drops a column from a table. Indexes and table constraints involving the column will be automatically dropped as well. You will need to say CASCADE if anything outside the table depends on the column, for example, foreign key references or views.

Reply:
@mageoffray Merci pour la remarque, j'ai enlevé les 2 drop index du `downgrade()` étant donné qu'il seront faits automatique par le drop table de `FutureOffer`.